app/uk/gov/hmrc/tai/config/ApplicationGlobal.scala

```diff
@@ -63,19 +70,21 @@ class TaiErrorHandler @Inject()(
       ))
   )
 
-  override def notFoundTemplate(implicit request: Request[_]): Html = standardErrorTemplate(
-    Messages("global.error.pageNotFound404.title"),
-    Messages("tai.errorMessage.heading"),
-    Messages(
-      "tai.errorMessage.frontend404",
-      Link
-        .toInternalPage(
-          url = routes.TaxAccountSummaryController.onPageLoad().url,
-          value = Some(Messages("tai.errorMessage.startAgain"))
-        )
-        .toHtml
-    )
-  )
+  override def onClientError(request: RequestHeader, statusCode: Int, message: String): Future[Result] = {
+    implicit val hc: HeaderCarrier =
+      HeaderCarrierConverter.fromHeadersAndSession(request.headers, Some(request.session))
+    implicit val req: Request[_] = Request(request, "")
+    statusCode match {
+      case play.mvc.Http.Status.NOT_FOUND =>
+        authorised() {
+          Future.successful(NotFound(views.html.page_not_found(isAuthenticated = true)))
```

Comment:
It's better to have two views here rather than pass in the bool for two different views